### ISSUE TITLE
New in-kernel (mac80211) driver for rtl8821au, rtl8811au and rtl8821ae chipsets.

### ISSUE THREAD

**Issue by morrownr** [aircrack-ng/rtl8812au#1152]
New in-kernel (mac80211) driver for rtl8821au, rtl8811au and rtl8821ae chipsets.

Information about testing is located at:

https://github.com/morrownr/8821au-20210708/issues/133

That is Issue 133 in the 8821au repo at github.com/morrow. That issue will be used for testing and reporting. The first message shows the location of the driver. The supported chipsets have been popular with Linux users for several years.

The intent is to upstream the driver to the Linux kernel once it is ready. There is nothing like Plug and Play adapters that work well. Please help if you can. I have been testing managed and monitor mode and the results are very good. We need to test rigorously.

HOT: All devs currently only have adapters with rtl8811au chips. We need Linux users with rtl8821au and rtl8821ce chips to test and report.

Additional developers are welcome. The more eyes we have on the project, the better. Work is also in progress on a driver for the rtl8812au chipset. It is not ready for testing yet.

Your help is appreciated.

@morrownr

**Comment by sec0ps** [comment]
Do you still need testers for this?

**Comment by morrownr** [comment]
Hi @sec0ps 

> Do you still need testers for this?

Yes

Go to the link in the first message above. It is the location where the work and testing are being coordinated. Recommend you read the entire thread to understand where we are. I look forward to seeing you there.

The first message once you are at the above link will give you the link to the repo where the driver is available for testing.

**Comment by kimocoder** [comment]
I've got all chipsets in the driver, 8812,  8811, 8814, 8821, will start soon.

First glance of 8821au, kernel taint / crash. will move to issue on rtw88 Larry Finger for debug/logs.

**Comment by morrownr** [comment]
Hi @kimocoder 

We are using the following issue at my 8821au repo for Project coordination and bug reporting for the 8821au:

morrownr/8821au-20210708#133

The reason has to do with ease of finding testers and some info needed to come out of that code anyway. Larry Finger has been nice enough to allow his rtw88 to serve as the code repo though.

> First glance of 8821au, kernel taint / crash.

We aren't seeing any crash reports right now. I'll be looking for your report.

> 8812

The intent is to move on to the 8812au as soon as soon as 8821au heads toward the kernel. 

We'll be looking for your reports and if you want to help with code, that would be great also.

**Comment by morrownr** [comment]
@kimocoder 

I was pondering why you might be seeing a crash:

Make sure you are compiling master. There are some other branches at Larry's rtw88 repo but all work has been consolidated in master at this point and that is what should be tested. While rtw88 contains several drivers, the one that is in work right now is rtl8821/11au. There is code there for rtl8812au but it is not ready for testing.

**Comment by kimocoder** [comment]
No, i had a bad system setup at that machine. I'll join the thread.
